Model/Menu/NodeRepository.php

```diff
@@ -44,7 +51,9 @@ public function __construct(
     public function save(NodeInterface $object)
     {
         try {
+            $isPageCacheClearable = $object->hasDataChanges();
```

Comment:
What do you think if we implement cache invalidation through plugins? this way we could decouple the code so external extensions can get rid of this feature disabling them [plugins]

I can see Magento core implements it this way, eg: https://github.com/magento/magento2/blob/2.4.0/app/code/Magento/Catalog/Plugin/Model/ResourceModel/Attribute/Save.php#L44

I think this applies to all the methods which invalidates cache (`save` and `delete` in menus and nodes repositories)

Reply:
The plugins are now implemented.